fedimint-api/src/db/mod.rs

```diff
@@ -13,17 +13,22 @@ pub mod mem_impl;
 
 pub use tests::test_db_impl;
 
-pub trait DatabaseKeyPrefixConst {
-    const DB_PREFIX: u8;
+pub trait DbKeyValue: DbKeyPrefix {
     type Key: DatabaseKey;
     type Value: DatabaseValue;
 }
 
-pub trait DatabaseKeyPrefix: Debug {
+// This trait is implemented for database key prefix enums.
+pub trait DbKeyPrefix {
+    type KeyPrefixType: std::convert::Into<u8>;
+    const KEY_PREFIX: Self::KeyPrefixType;
```

Comment:
I don't think it's possible to convert from enum to u8 given that `KEY_PREFIX` is `const`. The error should be:
```
error[[E0015]](https://doc.rust-lang.org/stable/error-index.html#E0015): cannot call non-const fn `<Prefix as Into<u8>>::into` in constants
```

Here's a Rust playground showing what I mean:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=da4c4f7c283bd3bfdfa3529b68cb6006

Do you know of a way around this error?

Reply:
I suppose each enum could implement its own `const fn as_u8(self) -> u8`.

Reply:
This could be solved in a much easier way imo: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7e516f1e05e5183ab0eb30d1a23162b5)

```rust
#[repr(u8)]
enum Foo {
    Bar = 0x01,
    Baz = 0x02,
}

const BAZ: u8 = Foo::Baz as u8;

fn main() {
    println!("BAZ = {}", BAZ);
}
```

Reply:
Thank you both, that's very clear code @elsirion , I'll go with that.